storm/src/test/storm/mesos/MesosCommonTest.java

```diff
@@ -0,0 +1,306 @@
+    nimbusConf.put(MesosCommon.MESOS_COMPONENT_NAME_DELIMITER, "--");
+    result = MesosCommon.getMesosComponentNameDelimiter(nimbusConf, info);
+    expectedResult = delimiter;
+    assertEquals(result, expectedResult);
+  }
+
+  @Test
+  public void testTaskId() throws Exception {
+    String nodeid = "nodeID1";
+    int port = 52;
+    // Regex format for timestamp is %d.%03d
+    String result = MesosCommon.taskId(nodeid, port);
+    String[] expectedElements = result.split("-");
+    // The taskId should be in at least three parts
+    assertTrue(3 >= expectedElements.length);
+    assertEquals(nodeid, expectedElements[0]);
+    String portAsString = "" + port + "";
+    assertEquals(portAsString, expectedElements[1]);
+    // The last part is timestamp number, ignore for now, will be different every time
+  }
+
+  @Test
+  public void testSupervisorId() throws Exception {
+    String nodeid = "nodeID1";
+    String topologyid = "t1";
+    String result = MesosCommon.supervisorId(nodeid, topologyid);
+    String expectedResult = nodeid + "-" + topologyid;
+    assertEquals(result, expectedResult);
+  }
+
+  @Test
+  public void testStartLogViewer() throws Exception {
+    // Test the default (true)
+    boolean result = MesosCommon.startLogViewer(conf);
+    assertTrue(result);
+    conf.put(MesosCommon.AUTO_START_LOGVIEWER_CONF, false);
+    result = MesosCommon.startLogViewer(conf);
+    assertTrue(!result);
+  }
+
+  @Test
+  public void testPortFromTaskId() throws Exception {
+    String taskid = "nodeid-22-time.stamp";
+    int result = MesosCommon.portFromTaskId(taskid);
+    int expectedResult = 22;
+
+    assertEquals(result, expectedResult);
+  }
+
+  @Test
+  public void testGetSuicideTimeout() throws Exception {
+    // Test default value
+    int result = MesosCommon.getSuicideTimeout(conf);
+    int expectedResult = MesosCommon.DEFAULT_SUICIDE_TIMEOUT_SECS;
+    assertEquals(result, expectedResult);
+
+    // Test explicit value
+    conf.put(MesosCommon.SUICIDE_CONF, 12);
+    result = MesosCommon.getSuicideTimeout(conf);
+    expectedResult = 12;
+    assertEquals(result, expectedResult);
+  }
+
+  @Test
+  public void testGetFullTopologyConfig() throws Exception {
+    Map nimbusConf = new HashMap<>();
+    nimbusConf.put("TEST_NIMBUS_CONFIG", 1);
+    nimbusConf.put("TEST_TOPOLOGY_OVERRIDE", 2);
+    Map topologyConf = new HashMap<>();
+    topologyConf.put("TEST_TOPOLIGY_CONFIG", 3);
+    topologyConf.put("TEST_TOPOLOGY_OVERRIDE", 4);
+    TopologyDetails info = new TopologyDetails("t1", topologyConf, new StormTopology(), 2);
+    Map result = MesosCommon.getFullTopologyConfig(nimbusConf, info);
+    Map expectedResult = new HashMap<>();
+    expectedResult.put("TEST_NIMBUS_CONFIG", 1);
+    expectedResult.put("TEST_TOPOLIGY_CONFIG", 3);
+    expectedResult.put("TEST_TOPOLOGY_OVERRIDE", 4);
+
+    assertEquals(result, expectedResult);
+  }
+
+  @Test
+  public void testTopologyWorkerCpu() throws Exception {
```

Comment:
One more test case that I can think off `conf.put(MesosCommon.WORKER_CPU_CONF, cpuConfig);`